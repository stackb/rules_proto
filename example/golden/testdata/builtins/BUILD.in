# gazelle:proto_rule proto_compile implementation stackb:rules_proto:proto_compile
# gazelle:proto_plugin cpp implementation builtin:cpp
# gazelle:proto_plugin java implementation builtin:java
# gazelle:proto_plugin closurejs implementation builtin:js:closure
# gazelle:proto_plugin commonjs implementation builtin:js:common
# gazelle:proto_plugin python implementation builtin:python
# gazelle:proto_plugin pyi implementation builtin:pyi
# gazelle:proto_plugin ruby implementation builtin:ruby
# gazelle:proto_plugin objc implementation builtin:objc
# gazelle:proto_language builtins rule proto_compile
# gazelle:proto_language builtins plugin cpp
# gazelle:proto_language builtins plugin java
# gazelle:proto_language builtins plugin closurejs
# gazelle:proto_language builtins plugin commonjs
# gazelle:proto_language builtins plugin python
# gazelle:proto_language builtins plugin pyi
# gazelle:proto_language builtins plugin ruby
# gazelle:proto_language builtins plugin objc
