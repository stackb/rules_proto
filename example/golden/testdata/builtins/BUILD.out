load("@rules_proto//proto:defs.bzl", "proto_library")
load("@build_stack_rules_proto//rules:proto_compile.bzl", "proto_compile")

# gazelle:proto_rule proto_compile implementation stackb:rules_proto:proto_compile
# gazelle:proto_plugin cpp implementation builtin:cpp
# gazelle:proto_plugin java implementation builtin:java
# gazelle:proto_plugin closurejs implementation builtin:js:closure
# gazelle:proto_plugin commonjs implementation builtin:js:common
# gazelle:proto_plugin python implementation builtin:python
# gazelle:proto_plugin pyi implementation builtin:pyi
# gazelle:proto_plugin ruby implementation builtin:ruby
# gazelle:proto_plugin objc implementation builtin:objc
# gazelle:proto_language builtins rule proto_compile
# gazelle:proto_language builtins plugin cpp
# gazelle:proto_language builtins plugin java
# gazelle:proto_language builtins plugin closurejs
# gazelle:proto_language builtins plugin commonjs
# gazelle:proto_language builtins plugin python
# gazelle:proto_language builtins plugin pyi
# gazelle:proto_language builtins plugin ruby
# gazelle:proto_language builtins plugin objc

proto_library(
    name = "test_proto",
    srcs = ["test.proto"],
    visibility = ["//visibility:public"],
)

proto_compile(
    name = "test_builtins_compile",
    outs = {"@build_stack_rules_proto//plugin/builtin:java": "test.srcjar"},
    options = {
        "@build_stack_rules_proto//plugin/builtin:closurejs": [
            "import_style=closure",
            "library=test",
        ],
        "@build_stack_rules_proto//plugin/builtin:commonjs": ["import_style=commonjs"],
    },
    outputs = [
        "Test.pbobjc.h",
        "Test.pbobjc.m",
        "test.js",
        "test.pb.cc",
        "test.pb.h",
        "test.srcjar",
        "test_pb.js",
        "test_pb.rb",
        "test_pb2.py",
        "test_pb2.pyi",
    ],
    plugins = [
        "@build_stack_rules_proto//plugin/builtin:closurejs",
        "@build_stack_rules_proto//plugin/builtin:commonjs",
        "@build_stack_rules_proto//plugin/builtin:cpp",
        "@build_stack_rules_proto//plugin/builtin:java",
        "@build_stack_rules_proto//plugin/builtin:objc",
        "@build_stack_rules_proto//plugin/builtin:pyi",
        "@build_stack_rules_proto//plugin/builtin:python",
        "@build_stack_rules_proto//plugin/builtin:ruby",
    ],
    proto = "test_proto",
)
